contracts/ParsecBridge.sol

```diff
@@ -547,6 +547,72 @@ contract ParsecBridge is Ownable {
     );
   }
 
+  function startBoughtExit(bytes32[] _proof, uint256 _oindex, bytes32[] signedData) public {
+
+    // root was submitted as period, check bridge was reciever
+    require(periods[_proof[0]].parent > 0);
+
+    // validate proof
+    bytes32 txHash;
+    bytes memory txData;
+    (, txHash, txData) = TxLib.validateProof(64, _proof);
+    // parse tx and use data
+    TxLib.Tx memory txn = TxLib.parseTx(txData);
+    TxLib.Output memory out = txn.outs[_oindex];
+    (uint256 buyPrice, bytes32 utxoIdSigned, address signer) = unpackSignedData(signedData);
+    // address sender = ecrecover(TxLib.getSigHash(txData), txn.ins[0].v, txn.ins[0].r, txn.ins[0].s);
+
+    require(out.owner == address(this), "Funds were not sent to bridge");
+    require(ecrecover(TxLib.getSigHash(txData), txn.ins[0].v, txn.ins[0].r, txn.ins[0].s) == signer, "Exit was not signed by owner");
+
+    uint256 exitableAt = Math.max256(periods[_proof[0]].timestamp + (2 * exitDuration), block.timestamp + exitDuration);
+    // bytes32 utxoIdProof = bytes32((_oindex << 120) | uint120(txHash));
+
+    require(bytes32((_oindex << 120) | uint120(txHash)) == utxoIdSigned, "The signed utxoid does not match the one in the proof");
+
+    uint256 priority = (exitableAt << 128) | uint128(utxoIdSigned);
+    require(out.value > 0);
+    require(exits[utxoIdSigned].amount == 0);
+    if (isNft(out.color)) {
+      nftExits[out.color].push(NftExit({ utxoId: utxoIdSigned, exitableAt: exitableAt }));
```

Comment:
i don't think we can sell NFTs this way, `tokens[out.color].addr.transferFrom(msg.sender, signer, buyPrice);` will not work. or are you thinking of a kind of NFT swap?

Reply:
You are right, I copy pasted the standard exit as my starting point and left this in for some reason. Will remove.